### PR TITLE
Michael Brady - Unit 2 - Sprint 4

### PR DESCRIPTION
**Pull request by mbrady4**

### CODE REVIEW

**Review by veritaem — COMMENTED** [review]
because the train and test and so different, and the train score is so low (i would expect in the 250-300 range), and the test is so high (would expect 250-300 again avg) i suspect you may be overfitting your model.  Have you tried making a more general model, like having a boolean flag for some variables (similar to a threshold)?

**Review by veritaem — COMMENTED** [review]
judging from the keyboard interrupt on the commit, id imagine the polynomial features weighed it down?  Its a computationally expensive process, and you want to pairplot to see what really needs to be polynomialized

**Review by veritaem — COMMENTED** [review]
Your pdp plots and things look very nice but If I may quibble with a couple things, specifically your 'model.fit' and your train test split?
* because the data in the df is ordered by date, and you just split it without shuffling, you lose a great deal of time series info (are more ppl buying bikes? Is pollution going up?).  train_test_split has a default shuffle, it may help?
* because you are estimating the best scoring model on your _training_ data, the best score is just the one that fits all of your already labeled points (which is why its max depth/max trees), which will lead to some overfitting and a lesser prediction score.  Have you tried making the model with a train score and predicting the val score to get your best model out that way?
